VideoLocker/src/main/java/org/edx/mobile/services/ViewPagerDownloadManager.java

```diff
@@ -0,0 +1,152 @@
+package org.edx.mobile.services;
+
+import android.content.res.Configuration;
+import android.os.Build;
+import android.os.Handler;
+import android.os.Looper;
+import android.view.View;
+
+import org.edx.mobile.base.MainApplication;
+import org.edx.mobile.logger.Logger;
+import org.edx.mobile.model.course.CourseComponent;
+import org.edx.mobile.util.WeakList;
+import org.edx.mobile.view.common.RunnableCourseComponent;
+import org.edx.mobile.view.common.TaskCallback;
+
+import java.util.List;
+
+/**
+ *  by default, ViewPager will populate UI for both current one-in-view and cached one-in-the-future.
+ *  in order to improve performance, we wont download it in parallel.
+ *  but it matters only for the initial load when user navigate from course outline page to detail page.
+ *
+ */
+public class ViewPagerDownloadManager implements TaskCallback {
+
+    protected final Logger logger = new Logger(getClass().getName());
+
+    public static ViewPagerDownloadManager instance = new ViewPagerDownloadManager();
+
+    private WeakList<RunnableCourseComponent> weakList;
```

Comment:
Maybe we should call this ``courseComponentList`` or something like that? ``weakList`` doesn't really tell me anything about what it does.